samcli/lib/providers/sam_function_provider.py

```diff
@@ -198,20 +208,31 @@ def _convert_lambda_function_resource(name, resource_properties, layers):  # pyl
         # CodeUri is set to "." in order to get code locally from current directory. AWS::Lambda::Function's ``Code``
         # property does not support specifying a local path
         codeuri = SamFunctionProvider.DEFAULT_CODEURI
+        inlinecode = None
         imageuri = None
         packagetype = resource_properties.get("PackageType", ZIP)
-
         if packagetype == ZIP:
-            codeuri = SamFunctionProvider._extract_lambda_function_code(resource_properties, "Code")
-            LOG.debug("Found Lambda function with name='%s' and CodeUri='%s'", name, codeuri)
+            if (
+                "Code" in resource_properties
+                and isinstance(resource_properties["Code"], dict)
+                and resource_properties["Code"].get("ZipFile") is not None
+            ):
+                inlinecode = resource_properties["Code"]["ZipFile"]
```

Comment:
It is interesting `InlineCode` can also be a zip file, just learned that from the doc